### PR TITLE
Changed variable names to get away from explicit mention of inverse c…

### PR DESCRIPTION
**Pull request by acliu**
…ovariance, as well as some tidying up of extraneous parts of Fisher et al. functions

### CODE REVIEW

File: hera_pspec/pspecdata.py [hunk 1]
```diff
@@ -376,129 +413,117 @@ def get_F(self, key1, key2, use_identity=False, true_fisher=False,
...
-    
-    def get_MW(self, F, mode='F^-1'):
+        G = np.zeros((self.Nfreqs, self.Nfreqs), dtype=np.complex)
+
+        R1 = self._R(key1)
+        R2 = self._R(key2)
+        
+        iR1Q, iR2Q = {}, {}
+        for ch in xrange(self.Nfreqs): # this loop is nchan^3
+            iR1Q[ch] = np.dot(R1, Q) # R_1 Q
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: I don't see where `Q` has been set in this function.

File: hera_pspec/pspecdata.py [hunk 2]
```diff
@@ -330,44 +359,52 @@ def q_hat(self, key1, key2, use_identity=False, use_fft=True,
...
+                RQR = np.einsum('ab,bc,cd', self._R(key1).T.conj(), self._Q[i], \
+                                                        self._R(key2)) # R Q R
+                qi = np.sum(self.x(k1).conj()*np.dot(RQR, self.x(k2)), axis=0)
                 q.append(qi)
             return np.array(q)
-    
-    def get_F(self, key1, key2, use_identity=False, true_fisher=False, 
-              taper='none'):
+
+    def get_G(self, key1, key2):
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: This gets rid of the FFT method, which is presumably faster. Do we want to take that hit?

Comment (acliu) [reply hera_pspec/pspecdata.py]: I don't think get_F ever had an FFT method. It was just the q_hat function that had the FFT option. I think it's possible to write a get_F function (or now get_G function) that uses the FFT method, but that would be a new implementation.

File: hera_pspec/pspecdata.py [hunk 3]
```diff
@@ -330,44 +359,52 @@ def q_hat(self, key1, key2, use_identity=False, use_fft=True,
...
         else:
             # Slow method, used to explicitly cross-check FFT code
             q = []
             for i in xrange(self.Nfreqs):
-                Q = self.get_Q(i, self.Nfreqs, taper=taper)
-                iCQiC = np.einsum('ab,bc,cd', iC1.T.conj(), Q, iC2) # C^-1 Q C^-1
-                qi = np.sum(self.x(k1).conj() * np.dot(iCQiC, self.x(k2)), axis=0)
+                RQR = np.einsum('ab,bc,cd', self._R(key1).T.conj(), self._Q[i], \
+                                                        self._R(key2)) # R Q R
+                qi = np.sum(self.x(k1).conj()*np.dot(RQR, self.x(k2)), axis=0)
```
[the remark this answers is not in the document]
Comment (acliu) [reply hera_pspec/pspecdata.py]: Yes, we were. Fixed; thanks!

File: hera_pspec/pspecdata.py [hunk 4]
```diff
@@ -291,28 +291,57 @@ def set_iC(self, d):
...
+        """
+
+        if R_matrix == "identity":
+            self._R = self.I
+        elif R_matrix == "iC":
+            self._R = self._iC
+        else:
+            self._R = R_matrix
+
+    def set_Qs(self, taper='none'):
```
[the remark this answers is not in the document]
Comment (acliu) [reply hera_pspec/pspecdata.py]: Yeah, it's tricky. I suppose we could put in an "if" statement so that if Q does not exist, it's computed. But then to do that, you would need to know the taper, which defeats the purpose of the reorganization.

Ok, I'm convinced. If the user is using PSpecData.pspec() anyway, they shouldn't have the opportunity to specify the taper in multiple locations anyway. I'll go back to the old way of doing things.

File: hera_pspec/pspecdata.py [hunk 5]
```diff
@@ -330,44 +359,52 @@ def q_hat(self, key1, key2, use_identity=False, use_fft=True,
...
-        icov_fn = self.I if use_identity else self.iC
-        
-        # Calculate C^-1 x_1 and C^-1 x_2
+        # Calculate R x_1 and R x_2
         #iC1x, iC2x = 0, 0
         #for _k in k1: iC1x += np.dot(icov_fn(_k), self.x(_k))
         #for _k in k2: iC2x += np.dot(icov_fn(_k), self.x(_k))
-        iC1x = np.dot(icov_fn(key1), self.x(key1))
-        iC2x = np.dot(icov_fn(key2), self.x(key2))
+        Rx1 = np.dot(self._R(key1), self.x(key1))
```
[the remark this answers is not in the document]
Comment (acliu) [reply hera_pspec/pspecdata.py]: Sorry, but I'm confused here. Why? Isn't iC, for example, self._iC?

File: hera_pspec/pspecdata.py [hunk 6]
```diff
@@ -330,44 +359,52 @@ def q_hat(self, key1, key2, use_identity=False, use_fft=True,
...
+            return np.conj(  np.fft.fftshift(_Rx1, axes=0).conj() 
+                           * np.fft.fftshift(_Rx2, axes=0) )
         else:
             # Slow method, used to explicitly cross-check FFT code
             q = []
             for i in xrange(self.Nfreqs):
-                Q = self.get_Q(i, self.Nfreqs, taper=taper)
-                iCQiC = np.einsum('ab,bc,cd', iC1.T.conj(), Q, iC2) # C^-1 Q C^-1
-                qi = np.sum(self.x(k1).conj() * np.dot(iCQiC, self.x(k2)), axis=0)
+                RQR = np.einsum('ab,bc,cd', self._R(key1).T.conj(), self._Q[i], \
```
[the remark this answers is not in the document]
Comment (acliu) [reply hera_pspec/pspecdata.py]: Fixed with the reversion to the old way of handling Q.

File: hera_pspec/pspecdata.py (hunk 1 above)
Comment (acliu) [reply hera_pspec/pspecdata.py]: Fixed; thank you!

File: hera_pspec/pspecdata.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (acliu) [reply hera_pspec/pspecdata.py]: Oh nevermind, I get it now.

File: hera_pspec/pspecdata.py [hunk 7]
```diff
@@ -613,7 +604,7 @@ def p_hat(self, M, q):
         """
         return np.dot(M, q)
 
-    def pspec(self, bls, weights='none', taper='none', verbose=False):
+    def pspec(self, bls, input_data_weight='identity',norm='I', taper='none', verbose=False):
```
Comment (philbull) [inline hera_pspec/pspecdata.py]: `input_data_weight` is a bit wordy -- can we get away with just calling it `data_weight`?

**Comment by philbull** [comment]
Changes approved and merged into `pspec_graduation`. Tests pass on my local machine -- TravisCI seems to be failing because of an upstream build error with `hera_cal`.
